src/test/java/org/sonarsource/sonarlint/ls/mediumtests/LanguageServerWithFoldersMediumTests.java

```diff
@@ -101,6 +101,9 @@ public static void resetSonarCloud() {
     System.clearProperty("sonarlint.internal.sonarcloud.websocket.url");
     System.clearProperty("sonarlint.internal.sonarcloud.url");
     System.clearProperty("sonarlint.internal.sonarcloud.api.url");
+    System.clearProperty("sonarlint.internal.sonarcloud.us.url");
```

Comment:
I could not make the test fail on my machine. I'm hoping this change will be enough. The other proposed change to always read a fresh value from System.getProperties will be more involved

Reply:
You can try like this, but I don't think it would be sufficient. The property is read in `ServerConnectionSettings` in a static field, which means whatever the property value is when the class is loaded, it will stay the same forever. Dependending on which test is executed first, it will influence the value for next ones. Another option would be to set the property once and for all, for all tests, maybe in the abstract class above all tests, or in another way. This would also have the good side effect to prevent any unwanted calls to prod